engine/src/main/java/com/datatorrent/stram/util/AbstractWritableAdapter.java

```diff
@@ -56,6 +63,9 @@ public void readFields(DataInput arg0) throws IOException
       ois.close();
     }
     catch (Exception e) {
+      final Path path = Files.createTempFile("dt-", ".ser");
+      logger.error("Failed to de-serialize {}. Writing raw data to {}.", this.getClass().getName(), path, e);
```

Comment:
That would be a much larger discussion that needs to consider all cases where data is written to disk and IMO does not belong into this PR.